float32 pos
float32 vel
float32 torq